### PR TITLE
Alert Rule Editor: new notes field

### PR DESCRIPTION
**Pull request by Bierchermuesli**
To address the complexities of managing numerous rules and the involvement of multiple administrators, I suggestan additional 'Notes' field within LibreNMS rule editor. This feature is designed to aid administrators in maintaining oversight and control. It provides a dedicated space for adding insights, comments, or audit-related information pertinent to specific rules.

image

- the `notes`attribute is also supported on the [Alert API](https://docs.librenms.org/API/Alerts) endpoint
- The `notes` attribute is also supported in the alert rule collection. Previously, rule names were missused for rule description and details, these details can now be placed in the notes field.  The "Device Down (SNMP unreachable)" is used as an example

I also replaced the super tiny advanced query text box before: 
image

to a textarea with allows multiline and monotext to avoid any typos (which was my case multiple times...) 
image

Certainly, here's a revised version:

Upcoming steps:
 -  convert the alert_rule.json (where it makes sense)
 - Implementing an audit possiblities for the alert rules, such as who enabled or disabled a specific rule. It hink this can be done with existing event log/logging framework. (long term idea..)

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [-] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply 15631`,
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: misc/db_schema.yaml [hunk 1]
```diff
@@ -2057,8 +2058,8 @@ syslog:
     - { Field: device_id, Type: 'int unsigned', 'Null': true, Extra: '' }
     - { Field: facility, Type: varchar(10), 'Null': true, Extra: '' }
     - { Field: priority, Type: varchar(10), 'Null': true, Extra: '' }
-    - { Field: level, Type: varchar(10), 'Null': true, Extra: '' }
```
Comment (PipoCanaja) [inline misc/db_schema.yaml:2060]: seems that you have something on the syslog side that should not be part of this PR

**Review by PipoCanaja — CHANGES_REQUESTED** [review]

File: misc/db_schema.yaml (hunk 1 above)
Comment (Bierchermuesli) [reply misc/db_schema.yaml:2060]: yeah. I noticed it by myself. thanks

File: database/migrations/2023_12_12_171400_alert_rule_note.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('alert_rules', function (Blueprint $table) {
+            $table->text('notes');
```
Comment (PipoCanaja) [inline database/migrations/2023_12_12_171400_alert_rule_note.php:15]: I would either allow "null" as a value (and adapt the code for it all along) or define a default value of `""` (empty string).

File: misc/db_schema.yaml [hunk 3]
```diff
@@ -101,6 +101,7 @@ alert_rules:
     - { Field: builder, Type: text, 'Null': false, Extra: '' }
     - { Field: proc, Type: varchar(80), 'Null': true, Extra: '' }
     - { Field: invert_map, Type: tinyint, 'Null': false, Extra: '', Default: '0' }
+    - { Field: notes, Type: text, 'Null': false, Extra: '' }
```
Comment (PipoCanaja) [inline misc/db_schema.yaml:104]: This will need to be generated again after migration is changed.

**Comment by PipoCanaja** [comment]
Hi @Bierchermuesli 
Looks good to me. Let me know what you think about the comments. 
The custom SQL change is a very good catch, got my head on this a couple of times.

File: includes/html/forms/create-alert-item.inc.php [hunk 4]
```diff
@@ -28,11 +28,8 @@
 $mute = $_POST['mute'];
 $invert = $_POST['invert'];
 $name = strip_tages($_POST['name']);
-if ($_POST['proc'] != '') {
-    $proc = $_POST['proc'];
-} else {
-    $proc = '';
-}
+$notes = isset($_POST['$notes']) ? $_POST['$notes'] : '';
```
Comment (PipoCanaja) [inline includes/html/forms/create-alert-item.inc.php:31]: We probably need strip tags here:
`$notes = isset($_POST['$notes']) ? strip_tags($_POST['$notes']) : '';`

File: includes/html/forms/create-alert-item.inc.php [hunk 5]
```diff
@@ -28,11 +28,8 @@
 $mute = $_POST['mute'];
 $invert = $_POST['invert'];
 $name = strip_tages($_POST['name']);
```
Comment (PipoCanaja) [inline includes/html/forms/create-alert-item.inc.php:30]: This looks like a mistake. should be `$name = strip_tags($_POST['name']);` i suppose

Comment (PipoCanaja) [reply includes/html/forms/create-alert-item.inc.php:30]: (i know it is not your code, but as you are around :D )

File: database/migrations/2023_12_12_171400_alert_rule_note.php [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::table('alert_rules', function (Blueprint $table) {
+            $table->text('notes')->default(null);
```
Comment (PipoCanaja) [inline database/migrations/2023_12_12_171400_alert_rule_note.php:15]: This should be
`$table->text('notes')->->nullable();` 
(and it will default to null without need to specify it)

**Review by PipoCanaja — CHANGES_REQUESTED** [review]

**Comment by PipoCanaja** [comment]
rerun tests

**Review by PipoCanaja — APPROVED** [review]
LGTM

**Comment by PipoCanaja** [comment]
Hi @Bierchermuesli 
Thanx for this PR !

**Comment by Bierchermuesli** [comment]
well, thank you @PipoCanaja for your help!
